oxide-auth-async/src/code_grant.rs

```diff
@@ -103,35 +103,52 @@ pub mod resource {
 
 pub mod access_token {
     use async_trait::async_trait;
-    use oxide_auth::code_grant::accesstoken::{
-        AccessToken, Request, BearerToken, Input, Output, Error, PrimitiveError,
-    };
-    use oxide_auth::primitives::{
-        grant::{Extensions, Grant},
-        registrar::RegistrarError,
+    use oxide_auth::{
+        endpoint::WebRequest,
+        primitives::{
+            grant::{Extensions, Grant},
+            registrar::RegistrarError,
+        },
+        code_grant::accesstoken::{AccessToken, BearerToken, Input, Output, Error, PrimitiveError},
     };
-
-    #[async_trait(?Send)]
-    pub trait Extension {
+    use crate::endpoint::access_token::WrappedRequest;
+
+    #[async_trait]
+    pub trait Extension<'a, R>
```

Comment:
Not to forget that it also propagates to `Extension` which it does not in the sync code. The reasoning was that extensions to the OAuth process should be independent from the actual request type, you shouldn't need to use any data that isn't part of the usual parameter passing that's standardized.